ip_country/src/ip_country.rs

```diff
@@ -273,4 +277,34 @@ Line 17: Invalid (AddrParseError(Ip)) IP address in CSV record: 'BOOGA'"#
 .to_string()
         );
     }
+
+    #[test]
+    fn write_error_test() {
+        let mut subject = BitQueue::new();
+        let output = &mut ByteArrayWriter::new();
+
+        subject.add_bits(0b11011, 5);
+        subject.add_bits(0b00111001110011100, 17);
+        subject.add_bits(0b1, 1);
+
+        output.reject_next_write(Error::new(ErrorKind::WriteZero, "Bad file Descriptor"));
+        let result = generate_country_data("ipv4_country_data", subject, output).unwrap_err();
```

Comment:
This is the "act" of the test.  It should stand insulated by the blank lines from both sides.